grails-app/services/com/unifina/service/PermissionService.groovy

```diff
@@ -482,9 +488,22 @@ class PermissionService {
 
 	@CompileStatic
 	private static String getResourcePropertyName(Object resource) {
-		// Derive property name from class name by turning it into camelCase
-		String simpleName = resource.getClass().getSimpleName()
-		return simpleName.substring(0, 1).toLowerCase() + simpleName.substring(1)
+		// Cannot derive name straight from resource.getClass() because of proxy assist objects!
+		if (resource instanceof Canvas) {
+			return "canvas"
+		} else if (resource instanceof Dashboard) {
+			return "dashboard"
+		} else if (resource instanceof Feed) {
+			return "feed"
+		} else if (resource instanceof ModulePackage) {
+			return "modulePackage"
+		} else if (resource instanceof Product) {
+			return "product"
+		} else if (resource instanceof Stream) {
+			return "stream"
+		} else {
+			throw new IllegalArgumentException("Unexpected resource class: " + resource)
+		}
```

Comment:
First feeling about this is that perhaps it could be simplified by just calling DomainClass.getResourcePropertyName() and let domain classes define their own resource property names. But I'm not sure if it actually simplifies anything?

Common base class implementation won't probably help here since its getClass() returns the base class name.

It would be a great idea to have a dynamically generated resource name because it would be refactor safe, but it might be too much work due to proxy assist stuff.

Reply:
I think one nice solution would be to have an interface `Permissionable` (or something better) that a domain class must implement to be able to use the Permission system. The `Permissionable` interface would require you implement method `getResourcePropertyName`. This would also grant type-safety.

Reply:
Made JIRA ticket out of this. See CORE-1292